kaldb/src/main/java/com/slack/kaldb/server/ArmeriaService.java

```diff
@@ -38,130 +36,106 @@
   }
 
-  private void initializeEventLoop(ServerBuilder sb) {
-    EventLoopGroup eventLoopGroup = EventLoopGroups.newEventLoopGroup(WORKER_EVENT_LOOP_THREADS);
-    EventLoopGroups.warmUp(eventLoopGroup);
-    sb.workerGroup(eventLoopGroup, true);
-  }
-
-  private void addCompression(ServerBuilder sb) {
-    sb.decorator(EncodingService.builder().newDecorator());
-  }
-
-  private void setTimeout(ServerBuilder sb) {
-    sb.requestTimeout(ARMERIA_TIMEOUT_DURATION);
-    // todo - on timeout explore including a retry-after header via exception handler
-    //  https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503
-    // sb.exceptionHandler()
-  }
-
-  private void addTracing(ServerBuilder sb) {
-    String endpoint = KaldbConfig.get().getTracingConfig().getZipkinEndpoint();
-    SpanHandler spanHandler = new SpanHandler() {};
-
-    if (!endpoint.isBlank()) {
-      LOG.info(String.format("Trace reporting enabled: %s", endpoint));
-      Sender sender = URLConnectionSender.create(endpoint);
-      spanHandler = AsyncZipkinSpanHandler.create(sender);
-    } else {
-      LOG.info("Trace reporting disabled");
+  public static class Builder {
+    private final String serviceName;
+    private final ServerBuilder serverBuilder;
+
+    public Builder(int port, String serviceName, PrometheusMeterRegistry prometheusMeterRegistry) {
+      this.serviceName = serviceName;
+      this.serverBuilder = Server.builder().http(port);
+
+      // initialize event loop
+      EventLoopGroup eventLoopGroup = EventLoopGroups.newEventLoopGroup(WORKER_EVENT_LOOP_THREADS);
+      EventLoopGroups.warmUp(eventLoopGroup);
+      this.serverBuilder.workerGroup(eventLoopGroup, true);
+
+      // timeout
+      this.serverBuilder.requestTimeout(ARMERIA_TIMEOUT_DURATION);
+
+      // compression
+      this.serverBuilder.decorator(EncodingService.builder().newDecorator());
+
+      // logging
+      this.serverBuilder.decorator(
+          LoggingService.builder()
+              // Not logging any successful response, say prom scraping /metrics every 30 seconds at
+              // INFO
+              .successfulResponseLogLevel(LogLevel.DEBUG)
+              .failureResponseLogLevel(LogLevel.ERROR)
+              // Remove the content to prevent blowing up the logs
+              .responseContentSanitizer((ctx, content) -> "truncated")
+              // Remove all headers to be sure we aren't leaking any auth/cookie info
+              .requestHeadersSanitizer((ctx, headers) -> DefaultHttpHeaders.EMPTY_HEADERS)
+              .newDecorator());
+
+      // management endpoints
+      this.serverBuilder
+          .service("/health", HealthCheckService.builder().build())
+          .service("/metrics", (ctx, req) -> HttpResponse.of(prometheusMeterRegistry.scrape()))
+          .serviceUnder("/docs", new DocService());
+
+      // tracing
+      String endpoint = KaldbConfig.get().getTracingConfig().getZipkinEndpoint();
```

Comment:
I am slowly converting all code to not use KaldbConfig.get directly. So, pass in the Tracing config explicitly and remove the dependency on `KaldbConfig.get()`. This will also make the unit tests easier to set up. 

Also, should we make the tracing config optional in the builder since it's not needed in unit tests?

Reply:
Ah, that is a good call-out. I went ahead and added another builder option that allows you to pass in the tracing endpoint when building ArmeriaService. If you don't provide it, it will automatically default to the non-reporting tracing like we do today.

For now it's still using the `KaldbConfig.get()` in `Kaldb.java` but directly passes the endpoint in to ArmeriaService - once we convert `Kaldb.java` it'll be an easy swap.